controllers/discovery/initDiscoveryTopicData.js

```diff
@@ -12,21 +13,22 @@ const InitDiscoveryTopicData = async (req, res) => {
   page = parseInt(page);
 
   const userId = req.userId;
+  const anonymousUser = await UsersFunction.findAnonymousUserId(User, userId, {raw: true});
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This line introduces a new asynchronous operation to fetch the anonymous user ID. It's important to handle potential errors that might occur during this operation. Consider wrapping this line in a try-catch block to handle any exceptions that might be thrown.
```javascript
try {
  const anonymousUser = await UsersFunction.findAnonymousUserId(User, userId, {raw: true});
} catch (error) {
  // Handle error
}
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->